tornjak-backend/pkg/agent/db/sqlite.go

```
@@ -76,13 +107,343 @@ func (db *LocalSqliteDb) GetAgents() (types.AgentInfoList, error) {
+			DomainName:   domainName,
+			ManagedBy:    managedBy,
+			PlatformType: platformType,
+			AgentsList:   agentsList,
+		})
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return types.ClusterInfoList{}, err
+	}
+	return types.ClusterInfoList{
+		Clusters: sinfos,
+	}, nil
+}
+
+// CreateClusterEntry takes in struct cinfo of type ClusterInfo.  If a cluster with cinfo.Name already registered, returns error.
+func (db *LocalSqliteDb) CreateClusterEntry(cinfo types.ClusterInfo) error {
+	ctx := context.Background()
+	tx, err := db.database.BeginTx(ctx, nil)
+	if err != nil {
+		return errors.New("Could not get context")
+	}
+
+	// CHECK existence of cluster
+	clusterExists, err := db.checkClusterExistence(ctx, tx, cinfo.Name)
+	if err != nil {
+		tx.Rollback()
+		return err
+	} else if clusterExists {
+		tx.Rollback()
+		return PostFailure{fmt.Sprintf("Error: cluster %v already exists", cinfo.Name)}
+	}
+
+	// INSERT cluster metadata
+	cmdInsert := "INSERT INTO clusters (name, domainName, managedBy, platformType) VALUES (?,?,?,?)"
+	statement, err := tx.PrepareContext(ctx, cmdInsert)
+	if err != nil {
+		tx.Rollback()
+		return SQLError{cmdInsert, err}
+	}
+	defer statement.Close()
+	_, err = statement.ExecContext(ctx, cinfo.Name, cinfo.DomainName, cinfo.ManagedBy, cinfo.PlatformType)
+	if err != nil {
+		tx.Rollback()
+		return SQLError{cmdInsert, err}
+	}
+
+	clusterID, err := db.getClusterID(ctx, tx, cinfo.Name)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	// ADD agents to cluster
+	err = db.addAgentsToCluster(ctx, tx, clusterID, cinfo.AgentsList)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	return tx.Commit()
+}
+
+// EditClusterEntry takes in struct cinfo of type ClusterInfo.  If cluster with cinfo.Name does not exist, throws error.
+func (db *LocalSqliteDb) EditClusterEntry(cinfo types.ClusterInfo) error {
+	ctx := context.Background()
+	tx, err := db.database.BeginTx(ctx, nil)
+	if err != nil {
+		return errors.New("Could not get context")
```

it should return the error, there is no guarantee that it failed because of context.